### ISSUE TITLE
New Style/YodaCondition false positive on !

### ISSUE THREAD

**Issue by zverok** [bbatsov/rubocop#4429]
```
$ rubocop -V
0.49.0 (using Parser 2.4.0.0, running on ruby 2.3.3 x86_64-linux)
```

Test file:
```ruby
class MyCollection
  def delete_if
    # some implementation
  end

  def keep_if
    delete_if { |val| !yield(val) }
  end
end
```

Rubocop output:
```
Inspecting 1 file
C

Offenses:

yoda.rb:7:23: C: Style/YodaCondition: Reverse the order of the operands !yield(val).
    delete_if { |val| !yield(val) }
                      ^^^^^^^^^^^
```

I believe it is nothing "Yoda" about this.

**Comment by smakagon** [comment]
@zverok thanks for report.
This issue been fixed by @bbatsov by this commit:
https://github.com/bbatsov/rubocop/commit/5700720cb92521e3a673e91ba517cc7884822826
I reproduced this issue with 0.49, but there is no offense in the latest master.
